src/js/libcs/OpDrawing.js

```diff
@@ -106,6 +106,183 @@ evaluator.fillcircle$2 = function(args, modifs) {
     return eval_helper.drawcircle(args, modifs, "F");
 };
 
+evaluator.drawarc$3 = function(args, modifs) {
+    return eval_helper.drawarc(args, modifs, "D");
+};
+
+evaluator.fillarc$3 = function(args, modifs) {
+    return eval_helper.drawarc(args, modifs, "F");
+};
+
+
+eval_helper.drawarc = function(args, modifs, df) {
+    var v1 = evaluateAndVal(args[0]);
+    var v2 = evaluateAndVal(args[1]);
+    var v3 = evaluateAndVal(args[2]);
+
+    var a = List.normalizeZ(csgeo.csnames[args[0].name].homog);
+    var b = List.normalizeZ(csgeo.csnames[args[1].name].homog);
+    var c = List.normalizeZ(csgeo.csnames[args[2].name].homog);
+    var abcdet = List.det3(a, b, c);
+
+    if (Math.abs(abcdet.value.real) > 1e-12) { // we have an arc, not segment
+        var con = geoOps._helper.ConicBy5(null, a, b, c, List.ii, List.jj);
+        var cen = geoOps._helper.CenterOfConic(con);
+        cen = List.normalizeMax(cen);
+
+        var zer = CSNumber.real(0);
+
+        // move center of conic to origin
+        var mat = List.turnIntoCSList([
+            List.turnIntoCSList([cen.value[2], zer, CSNumber.neg(cen.value[0])]),
+            List.turnIntoCSList([zer, cen.value[2], CSNumber.neg(cen.value[1])]),
+            List.turnIntoCSList([zer, zer, cen.value[2]])
+        ]);
+        var aa = List.normalizeZ(General.mult(mat, a));
+        var bb = List.normalizeZ(General.mult(mat, b));
+        var cc = List.normalizeZ(General.mult(mat, c));
+
+
+        // get angles of A and C 
+        var startAngle = -Math.atan2(aa.value[1].value.real, aa.value[0].value.real);
+        var endAngle = -Math.atan2(cc.value[1].value.real, cc.value[0].value.real);
+
+        cen = List.normalizeZ(cen);
+        var arcDist = List.abs(List.sub(a, cen));
```

Comment:
I wonder whether here could be any reason to compute the radius directly from the matrix, instead of via a distance as you did. Probably not, just thinking out loud. This *is* the radius, right? Because the term `arcDist` sounds so strange if there is such a simple term instead.